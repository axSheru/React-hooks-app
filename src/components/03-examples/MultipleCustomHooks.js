import React from 'react';
import { useFetch } from '../../hooks/useFetch';
import { useCounterGenerico } from '../../hooks/useCounterGenerico';
import '../../index.css';

export const MultipleCustomHooks = () => {

    const { counter, increment } = useCounterGenerico(1);

    const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${ counter }`);

    const { author, quote } = !!data && data[0];

    return (
        <>
            <h1>Breaking Bad Quotes</h1>
            <hr />

            {
                loading
                ?
                    (
                        <div className='alert alert-info text-center'>Loading...</div>
                    )
                :
                    (
                        <>
                            <blockquote className='blockquote text-right'>
                                <p className='mb-0'>{ quote }</p>
                                <footer className='blockquote-footer'>{ author }</footer>
                            </blockquote>
                            <button
                                className='btn btn-outline-primary'
                                onClick={ increment }
                            >
                                Siguiente quote
                            </button>
                        </>
                    )
            }
            
        </>
    )
}
