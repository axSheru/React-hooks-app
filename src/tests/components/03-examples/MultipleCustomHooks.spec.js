import { shallow } from "enzyme";
import { MultipleCustomHooks } from "../../../components/03-examples/MultipleCustomHooks";

describe('Pruebas en el componente MultipleCustomHooks.', () => {

    test('debe de hacer match con el snapshot.', () => {

        const wrapper = shallow( <MultipleCustomHooks /> );

        expect( wrapper ).toMatchSnapshot();

    });

});