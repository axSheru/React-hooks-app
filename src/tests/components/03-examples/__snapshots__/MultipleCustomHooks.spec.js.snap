// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente MultipleCustomHooks. debe de hacer match con el snapshot. 1`] = `
<Fragment>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
</Fragment>
`;
